### ISSUE TITLE
Suggestion: Enhance XRONOS with Model-Based Pleiades Matching Strategy

### ISSUE THREAD

**Issue by MartinHinz** [xronos-ch/xronos.rails#350]
**Description:**

To enable the integration of the Pleiades dataset as LOD, I propose introducing a model-based solution within XRONOS. By implementing a matching method directly within either the `Site` model or the `PleiadesItem` model, we can establish an efficient way to link XRONOS site names with Pleiades data.

---

**Proposed Enhancement:**

### 1. Introduce a `PleiadesItem` Model
Add a model named `PleiadesItem` to store Pleiades site names and their corresponding IDs. This will provide the necessary structure for matching with the `Site` model.

**Example model schema:**
```ruby
class PleiadesItem < ApplicationRecord
  validates :name, presence: true
  validates :pleiades_id, presence: true
end
```

### 2. Periodic Synchronization of Pleiades Data

Implement a rake task to populate and update the PleiadesItem model with the latest data from the Pleiades name_index.json. This ensures the data remains current.

Example task:

```ruby
namespace :pleiades do
  desc "Sync Pleiades data with XRONOS"
  task sync: :environment do
    require 'open-uri'
    require 'json'

    url = 'https://raw.githubusercontent.com/ryanfb/pleiades-geojson/gh-pages/name_index.json'
    pleiades_data = JSON.parse(URI.open(url).read)

    pleiades_data.each do |name, id|
      PleiadesItem.find_or_create_by(name: name.strip.downcase, pleiades_id: id)
    end
  end
end
```

#### a) Method in the Site Model

Add a method in the Site model to find a corresponding PleiadesItem for a site by comparing names:

Example:

```ruby
class Site < ApplicationRecord
  def match_to_pleiades
    PleiadesItem.where("name ILIKE ?", "%#{self.name.strip.downcase}%").first
  end
end
```

Usage:

```ruby
site = Site.find(1)
match = site.match_to_pleiades
if match
  puts "Site '#{site.name}' matches PleiadesItem '#{match.name}' with ID #{match.pleiades_id}"
else
  puts "No match found for site '#{site.name}'"
end
```

#### b) Method in the PleiadesItem Model

Alternatively, add a method in the PleiadesItem model to find all Site entries matching a specific Pleiades name:

**Example:**

```ruby
class PleiadesItem < ApplicationRecord
  def match_sites
    Site.where("name ILIKE ?", "%#{self.name.strip.downcase}%")
  end
end
```

**Usage:**

```ruby
pleiades_item = PleiadesItem.find(1)
matching_sites = pleiades_item.match_sites
matching_sites.each do |site|
  puts "PleiadesItem '#{pleiades_item.name}' matches Site '#{site.name}'"
end
```

### 4. Further Task

* Add fuzzy matching capabilities using gems like amatch or fuzzy_match to improve match accuracy.
* Store match results by adding a pleiades_id field to a link table similar to the wikidata approach.

**Benefits:**
* Embeds the matching logic directly into the relevant models, keeping the code clean and cohesive.
* Supports efficient and reusable methods for site-to-Pleiades matching.
* Enables researchers to link XRONOS sites with Pleiades data seamlessly.

**Potential Issues:**
* Matching accuracy depends on data quality and name standardization.
* Requires maintenance of the PleiadesItem model and periodic synchronization.

I look forward to feedback on this proposal!

**Comment by joeroe** [comment]
I don't think this approach matches our current architecture for linked open data or external API interaction very well. For an alternative approach that could be copied here, see #485. But Wikidata already indexes many of these, so if we get matches to it right, we'll get most of the others for free.
